### PR TITLE
ci: Perform canary releases from the (protected) canary branch

### PR DESCRIPTION
**Pull request by zioroboco**
Causes releasable changes pushed to the branch `canary` to publish those changes to the npm registry under a pre-release canary version. Merging into canary release branches will still require a reviewed pull request, and changes will be versioned according to conventional commit on the canary branch.

For example, if a feature branch with commit `feat: Some changes` modifying a package currently at version `1.0.0` is merged into `canary` (via an approved pull request), the published package version will be `1.1.0-canary.0`.

After merging each of these PRs, the `canary` branch will automatically be reset to the current `master`.

These packages will be available for use in test environments by specifying the released package version exactly, or by using e.g. `yarn add @kaizen/component-library@canary` to get the latest canary release of the specified package.

Note that package versions released from this workflow should not be used in production.

### CODE REVIEW

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -51,6 +61,7 @@ steps:
 
   - name: ":seedling: Publish: ${KAIZEN_DOMAIN_NAME}${KAIZEN_BASE_PATH}"
     command: ".buildkite/scripts/publish.sh"
+    branches: "!canary/*"
```
Comment (zioroboco) [inline .buildkite/pipeline.yaml:64]:
```suggestion
    # canary release branches are not able to update dev.cultureamp.design
    branches: "!canary/*"
```

File: .buildkite/scripts/release.sh [hunk 2]
```diff
@@ -47,24 +47,49 @@ release() {
...
+    --message "chore: release [skip ci]"
+
   # Publish any package versions which are not already present on npm
   yarn lerna publish from-package --yes
 }
 
+release_canary() {
+  yarn install --frozen-lockfile
+
+  CANARY_LABEL=$(echo "$BUILDKITE_BRANCH" | sed 's/[^0-9A-Za-z-]/-/g')
```
Comment (zioroboco) [inline .buildkite/scripts/release.sh:59]:
```suggestion
  # Replace illegal characters according to semver prerelease versioning
  # https://semver.org/#spec-item-9
  CANARY_LABEL=$(echo "$BUILDKITE_BRANCH" | sed 's/[^0-9A-Za-z-]/-/g')
```

**Review by ActuallyACat — APPROVED** [review]
:shipit:

File: README.md [hunk 3]
```diff
@@ -60,6 +60,13 @@ While the main site is located at <https://cultureamp.design>, all branches in t
 
 For example, pushing the branch `louis/more-glitter` would make its documentation and Storybook build available at `dev.cultureamp.design/louis/more-glitter`.
 
+## Canary Releases 
+Canary releases are a great way to reduce the risk of your changes. Use canary releases when you're working on a significant refactor, experimenting with new technology, or making other large scale changes. 
+
+Canary branches are protected, and can only be created by the admins of this repo. Once you have a canary branch set up, any subsequent PR + merge into that branch will release a package. As an example, a repo admin
```
Comment (jasononeil) [inline README.md:66]: Can a non-admin create a `canary/my-successful-hack` branch?

Or does it being protected protect against pushes of new branches

Comment (zioroboco) [reply README.md:66]: *sigh*

**Comment by ActuallyACat** [comment]
LGTM ✅

File: .buildkite/scripts/release.sh [hunk 4]
```diff
@@ -62,9 +70,31 @@ main() {
...
+  elif [ "$BUILDKITE_BRANCH" = canary ]; then
+
+    echo "Branch: canary"
+
+    echo "Releasing packages..."
+    release_canary
+
+    echo "Resetting canary branch..."
+    git reset --hard master
+    git push --force
```
Comment (jasononeil) [inline .buildkite/scripts/release.sh:93]: Beautiful, that's a nice little touch 👍

**Review by jasononeil — APPROVED** [review]

**Pull request merged**